pages/tool.tsx

```diff
@@ -1,49 +1,239 @@
 import React from "react";
-import { Text, Spinner, Heading, AspectRatio } from "@chakra-ui/react";
+import { Box, Center, Flex, Spacer, Spinner, Text } from "@chakra-ui/react";
 import axios from "axios";
 import useSWR from "swr";
-
 import { useRouter } from "next/router";
+import "react-responsive-carousel/lib/styles/carousel.min.css"; // requires a loader
+import { Carousel } from "react-responsive-carousel";
+import ReactPlayer from "react-player";
+
+import { ToolLayout } from "@components";
+import Footer from "src/pages/tool/Footer";
+import Header from "src/pages/tool/Header";
+import ModuleProgressSection from "src/pages/tool/ModuleProgressSection";
+import ResourcesAccordion from "src/pages/tool/ResourcesAccordion";
+import StartModuleSection from "src/pages/tool/StartModuleSection";
+import ToolCard from "src/pages/tool/ToolCard";
+import { getIsDesktop } from "src/utils/media/mediaHelpers";
+import { Page } from "types/Page";
 
-const fetcher = async (url) => {
+const fetcher = async (url: string) => {
     const response = await axios({
         method: "GET",
         url,
     });
     return response.data;
 };
 
-const Module: React.FC = () => {
+const relatedToolsFetcher = async (relatedToolsIDs: string[]) => {
+    const data = [];
+    for (const relatedToolId of relatedToolsIDs) {
+        if (relatedToolId !== "") {
+            const response = await axios({
+                method: "GET",
+                url: `/api/tool/${relatedToolId}`,
+            });
+            data.push(response.data);
+        }
+    }
+    return data;
+};
+
+const Tool: Page = () => {
+    const [isDesktop] = getIsDesktop();
+    const progress = 100;
+
     const router = useRouter();
     const { id } = router.query;
-
     const fetchURL = id ? `/api/tool/${id}` : null;
     const { data, error } = useSWR(() => fetchURL, fetcher);
-    if (error) return <div>An error has occurred.</div>;
-    if (!data) return <Spinner color="brand.lime" size="xl" />;
-
-    return (
-        <>
-            <Heading> {data.title} </Heading>
-
-            <AspectRatio maxW="100%" maxH={290} ratio={1}>
-                <iframe title="Toolkit" src={data.video} allowFullScreen />
-            </AspectRatio>
-
-            <br />
-
-            <Text textAlign="left"> {data.description} </Text>
+    const { data: relatedToolsData, error: relatedToolsError } = useSWR(
+        data ? [data.relatedToolsIDs] : null,
+        relatedToolsFetcher,
+    );
 
-            <br />
+    if (error || relatedToolsError) {
```

Comment:
If an error occurs with the related tools, we should still render as much as we can. Consider moving the error handling for related tools to be just under the related tools section

Reply:
move to conditional rendering on line 113